### PR TITLE
Bump all applications

### PR DESCRIPTION
**Pull request by SwayStar123**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following apps have been updated to forc 0.60.0, fuel core 0.26.0, fuels 0.62.0

- [x] airdrop
- [x] AMM
- [x] counter-script
- [x] DAO
- [x] english-auction
- [x] escrow
- [x] fractional-NFT
- [x] fundraiser
- [ ] multisig-wallet (this will be updated in a different pr)
- [x] name-registry
- [x] native-asset
- [x] NFT
- [x] oracle
- [x] OTC-swap-predicate
- [x] TicTacToe
- [x] timelock

### CODE REVIEW

**Review by matt-user — COMMENTED** [review]
overall looks good.  Just some small suggestions

File: DAO/DAO-contract/src/main.sw [hunk 1]
```diff
@@ -44,7 +45,7 @@ storage {
     /// The initialization state of the contract.
     state: State = State::NotInitialized,
     /// Contract Id of the governance asset
-    asset: AssetId = AssetId::base_asset_id(),
+    asset: AssetId = AssetId::from(ZERO_B256),
```
[the remark this answers is not in the document]
Comment (SwayStar123) [reply DAO/DAO-contract/src/main.sw:48]: cant use non static function to initialize storage

File: AMM/swap-exact-input/tests/cases/revert.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use crate::utils::{expected_and_actual_output, expected_swap_output, setup};
 use fuels::prelude::AssetId;
-use test_utils::{data_structures::SwapParameters, setup::scripts::contract_instances};
+use test_utils::{data_structures::{SwapParameters, NUMBER_OF_ASSETS}, setup::scripts::contract_instances};
```
[the remark this answers is not in the document]
Comment (SwayStar123) [reply AMM/swap-exact-input/tests/cases/revert.rs:3]: i dont know, but its used in the last test in that file

**Review by bitzoic — APPROVED** [review]
Change what you need to get this merged and make issues for the rest to be addressed separately.

**Review by matt-user — APPROVED** [review]

File: AMM/fuel-toolchain.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [toolchain]
-channel = "nightly-2023-12-14"
+channel = "nightly-2024-05-28"
 
 [components]
-forc = "0.49.1"
-fuel-core = "0.22.0"
+forc = "0.60.0+nightly.20240528.4fe6f1ed51"
```
[the remark this answers is not in the document]
Comment (SwayStar123) [reply AMM/fuel-toolchain.toml:5]: yeah it needs a bugfix thats not released

**Review by matt-user — APPROVED** [review]

**Comment by K1-R1** [comment]
Re-running CI as the DAO app was skipped, otherwise lgtm

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**